File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        min_len = float("inf")
        cur = 0
        for right in range(len(nums)):
            cur += nums[right]
            while cur >= target:
                min_len = min(min_len, right - left + 1)
                cur -= nums[left]
                left += 1
        return int(min_len) if min_len < float('inf') else 0

File: Data Structures & Algorithms/majority-element-ii/submission-2.go
func majorityElement(nums []int) []int {
    if len(nums) == 0 {
        return nil
    }
    freq := make(map[int]int)
    out := make([]int, 0)
    target := len(nums) / 3
    for _, n := range nums {
        freq[n]++
    }
    for n, count := range freq {
        if count > target {
            out = append(out, n)
        }
    }
    return out
}
